src/document/mod.rs

```diff
@@ -84,113 +116,225 @@ impl DocumentEncryptOpts {
+    /// let encrypted = sdk.document_encrypt(data, &DocumentEncryptOpts::default()).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_encrypt(
+        &self,
+        document_data: &[u8],
+        encrypt_opts: &DocumentEncryptOpts,
+    ) -> Result<DocumentEncryptResult>;
+
+    /// Decrypts an IronCore encrypted document.
+    ///
+    /// Returns details about the document as well as its decrypted bytes.
+    ///
+    /// # Arguments
+    /// - `encrypted_document` - Bytes of encrypted document. These should be the same bytes returned from
+    /// [document_encrypt()](trait.DocumentOps.html#tymethod.document_encrypt).
+    ///
+    /// # Errors
+    /// Fails if passed malformed data or if the calling user does not have sufficient access to the document.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let encrypted_data: Vec<u8> = vec![];
+    /// let decrypted_document = sdk.document_decrypt(&encrypted_data).await?;
+    /// let decrypted_data = decrypted_document.decrypted_data();
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_decrypt(&self, encrypted_document: &[u8]) -> Result<DocumentDecryptResult>;
+
+    /// Lists all of the IronCore encrypted documents that the calling user has been
+    /// granted access to.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # use ironoxide::document::DocumentListMeta;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// let document_data = sdk.document_list().await?;
+    /// let documents: Vec<DocumentListMeta> = document_data.result().to_vec();
+    /// # Ok(())
+    /// # }
     async fn document_list(&self) -> Result<DocumentListResult>;
 
-    /// Get the metadata for a specific document given its ID.
+    /// Returns the metadata for an IronCore encrypted document.
     ///
     /// # Arguments
     /// - `id` - Unique ID of the document to retrieve
     ///
-    /// # Returns
-    /// `DocumentMetadataResult` with details about the requested document.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// let document_data = sdk.document_get_metadata(&document_id).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn document_get_metadata(&self, id: &DocumentId) -> Result<DocumentMetadataResult>;
 
-    /// Attempt to parse the document ID out of an encrypted document.
+    /// Returns the document ID from the bytes of an IronCore encrypted document.
+    ///
+    /// This is the same ID returned by `DocumentEncryptResult.id()`.
```

Comment:
```suggestion
    /// This is the same ID returned by `DocumentEncryptResult.id()` or `DocumentMetadataResult.id()`.
```

Reply:
While the addition is true, I was only calling out that it's the same as `DocumentEncryptResult.id()` to show that someone who just encrypted a file shouldn't call this, since they already have access to that data. Maybe I need to explicitly call that out.

So I think I'd rather leave out the addition to keep it focused on the intent of it.

Reply:
Yep, that's fine.